lib/task_helpers/development/replication.rb

```diff
@@ -0,0 +1,128 @@
+
+          # TODO: We have the technology to watch for this and report when it's all good or bad
+          puts "Local replication is setup... try checking for users in all regions: psql -U root development_replication_99 -c \"select id from users;\""
+        ensure
+          restore
+        end
+
+        def teardown
+          REMOTES.each do |r|
+            teardown_global_subscription_for_region(r)
+            teardown_remote_publication(r)
+          end
+        end
+
+        def setup_remote(region)
+          create_region(region)
+          configure_remote_region(region)
+        end
+
+        def setup_global(region)
+          create_region(region)
+          configure_global_region(region)
+        end
+
+        def database(region)
+          "#{DB_PREFIX}_#{region}"
+        end
+
+        # Example: DATABASE_URL='postgres://root:smartvm@localhost:5432/development_replication_99'
+        def database_url(region)
+          "postgres://#{PG_USER}:#{PG_PASS}@#{PG_HOST}:#{PG_PORT}/#{database(region)}"
+        end
+
+        def guid_file
+          GUID_FILE
+        end
+
+        def backup_guid
+          BACKUP_GUID
+        end
+
+        def configure_global_region_script
+          subs = []
+          REMOTES.each do |r|
+            subs << PglogicalSubscription.new(:host => PG_HOST, :port => PG_PORT, :user => PG_USER, :dbname => database(r), :password => PG_PASS)
+          end
+          MiqPglogical.save_global_region(subs, [])
+        end
+
+        private
+
+        def command_environment(region)
+          {"REGION" => region.to_s, "RAILS_ENV" => "development", "DATABASE_URL" => database_url(region)}
+        end
+
+        def configure_global_region(region)
+          run_command("bin/rails r 'TaskHelpers::Development::Replication.configure_global_region_script'", env: command_environment(region))
+          run_command("psql -U #{PG_USER} #{database(region)} -c 'select * from pg_subscription;'")
+        end
+
+        def configure_remote_region(region)
+          run_command("bin/rails r 'MiqRegion.replication_type= :remote'", env: command_environment(region))
+          run_command("psql -U #{PG_USER} #{database(region)} -c 'select * from pg_publication;'")
+        end
+
+        def create_region(region)
+          run_command("DISABLE_DATABASE_ENVIRONMENT_CHECK='true' bin/rake evm:db:region", env: command_environment(region))
+          run_command("bin/rails r 'EvmDatabase.seed_primordial'", env: command_environment(region))
+        ensure
+          FileUtils.rm_f(guid_file)
+        end
+
+        def run_command(command, raise_on_error: true, env: {})
+          puts "+" * 50
+          puts "Running: #{command.inspect} with env: #{env.inspect}..."
+          puts AwesomeSpawn.run!(command, env: env).output
+        rescue AwesomeSpawn::CommandResultError => err
+          raise if raise_on_error
+          puts "Error skipped: #{err.to_s}"
+        ensure
+          puts "-" * 50
```

Comment:
I'm not a huge fan of the output but probably need others to try it out and offer better suggestions... I'm too deep into this and can't get out.  It looks fine to me.  🤣 

```
joerafaniello@Joes-MacBook-Pro-2 manageiq % be rake development:replication:setup
** ManageIQ master, codename: Morphy
++++++++++++++++++++++++++++++++++++++++++++++++++
Running: "DISABLE_DATABASE_ENVIRONMENT_CHECK='true' bin/rake evm:db:region" with env: {"REGION"=>"1", "RAILS_ENV"=>"development", "DATABASE_URL"=>"postgres://root:smartvm@localhost:5432/development_replication_1"}...
Resetting development database to region 1...
** ManageIQ master, codename: Morphy
Initializing region and database...
--------------------------------------------------
++++++++++++++++++++++++++++++++++++++++++++++++++
Running: "bin/rails r 'EvmDatabase.seed_primordial'" with env: {"REGION"=>"1", "RAILS_ENV"=>"development", "DATABASE_URL"=>"postgres://root:smartvm@localhost:5432/development_replication_1"}...
** ManageIQ master, codename: Morphy
--------------------------------------------------
++++++++++++++++++++++++++++++++++++++++++++++++++
Running: "bin/rails r 'MiqRegion.replication_type= :remote'" with env: {"REGION"=>"1", "RAILS_ENV"=>"development", "DATABASE_URL"=>"postgres://root:smartvm@localhost:5432/development_replication_1"}...
** ManageIQ master, codename: Morphy
--------------------------------------------------
++++++++++++++++++++++++++++++++++++++++++++++++++
Running: "psql -U root development_replication_1 -c 'select * from pg_publication;'" with env: {}...
 pubname | pubowner | puballtables | pubinsert | pubupdate | pubdelete
---------+----------+--------------+-----------+-----------+-----------
 miq     |    16385 | f            | t         | t         | t
(1 row)

--------------------------------------------------
++++++++++++++++++++++++++++++++++++++++++++++++++
Running: "DISABLE_DATABASE_ENVIRONMENT_CHECK='true' bin/rake evm:db:region" with env: {"REGION"=>"2", "RAILS_ENV"=>"development", "DATABASE_URL"=>"postgres://root:smartvm@localhost:5432/development_replication_2"}...
Resetting development database to region 2...
** ManageIQ master, codename: Morphy
Initializing region and database...
--------------------------------------------------
++++++++++++++++++++++++++++++++++++++++++++++++++
Running: "bin/rails r 'EvmDatabase.seed_primordial'" with env: {"REGION"=>"2", "RAILS_ENV"=>"development", "DATABASE_URL"=>"postgres://root:smartvm@localhost:5432/development_replication_2"}...
** ManageIQ master, codename: Morphy
--------------------------------------------------
++++++++++++++++++++++++++++++++++++++++++++++++++
Running: "bin/rails r 'MiqRegion.replication_type= :remote'" with env: {"REGION"=>"2", "RAILS_ENV"=>"development", "DATABASE_URL"=>"postgres://root:smartvm@localhost:5432/development_replication_2"}...
** ManageIQ master, codename: Morphy
--------------------------------------------------
++++++++++++++++++++++++++++++++++++++++++++++++++
Running: "psql -U root development_replication_2 -c 'select * from pg_publication;'" with env: {}...
 pubname | pubowner | puballtables | pubinsert | pubupdate | pubdelete
---------+----------+--------------+-----------+-----------+-----------
 miq     |    16385 | f            | t         | t         | t
(1 row)

--------------------------------------------------
++++++++++++++++++++++++++++++++++++++++++++++++++
Running: "DISABLE_DATABASE_ENVIRONMENT_CHECK='true' bin/rake evm:db:region" with env: {"REGION"=>"99", "RAILS_ENV"=>"development", "DATABASE_URL"=>"postgres://root:smartvm@localhost:5432/development_replication_99"}...
Resetting development database to region 99...
** ManageIQ master, codename: Morphy
Initializing region and database...
--------------------------------------------------
++++++++++++++++++++++++++++++++++++++++++++++++++
Running: "bin/rails r 'EvmDatabase.seed_primordial'" with env: {"REGION"=>"99", "RAILS_ENV"=>"development", "DATABASE_URL"=>"postgres://root:smartvm@localhost:5432/development_replication_99"}...
** ManageIQ master, codename: Morphy
--------------------------------------------------
++++++++++++++++++++++++++++++++++++++++++++++++++
Running: "bin/rails r 'TaskHelpers::Development::Replication.configure_global_region_script'" with env: {"REGION"=>"99", "RAILS_ENV"=>"development", "DATABASE_URL"=>"postgres://root:smartvm@localhost:5432/development_replication_99"}...
** ManageIQ master, codename: Morphy
--------------------------------------------------
++++++++++++++++++++++++++++++++++++++++++++++++++
Running: "psql -U root development_replication_99 -c 'select * from pg_subscription;'" with env: {}...
 subdbid |        subname        | subowner | subenabled |                                                              subconninfo                                                              |      subslotname      | subsynccommit | subpublications
---------+-----------------------+----------+------------+---------------------------------------------------------------------------------------------------------------------------------------+-----------------------+---------------+-----------------
 2389375 | region_1_subscription |    16385 | t          |  fallback_application_name='bin/rails' dbname='development_replication_1' host='localhost' user='root' password='smartvm' port='5432' | region_1_subscription | off           | {miq}
 2389375 | region_2_subscription |    16385 | t          |  fallback_application_name='bin/rails' dbname='development_replication_2' host='localhost' user='root' password='smartvm' port='5432' | region_2_subscription | off           | {miq}
(2 rows)

--------------------------------------------------
Local replication is setup... try checking for users in all regions: psql -U root development_replication_99 -c "select id from users;"
```